### PR TITLE
[Mono.Profiler.Aot] Add profile writer

### PR DESCRIPTION
**Pull request by radekdoulik**
It writes the profile data to given stream. It comes from
https://github.com/radekdoulik/aotprofile-tool

### CODE REVIEW

File: mcs/class/Mono.Profiler.Log/Mono.Profiler.Aot/ProfileBase.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+namespace Mono.Profiler.Aot {
+	public abstract class ProfileBase {
+
+		protected enum RecordType {
```
Comment (marek-safar) [inline mcs/class/Mono.Profiler.Log/Mono.Profiler.Aot/ProfileBase.cs]: We don't want to expose them externally, all these should be internal or private protected

File: mcs/class/Mono.Profiler.Log/Mono.Profiler.Aot/ProfileWriter.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+			WriteString (m.Signature);
+		}
+
+		void WriteRecord (RecordType rt, int value)
+		{
+			s.WriteByte ((byte)rt);
+			WriteInt32 (value);
+		}
+
+		public void Write ()
```
Comment (marek-safar) [inline mcs/class/Mono.Profiler.Log/Mono.Profiler.Aot/ProfileWriter.cs]: Move `ProfileData data` as the argument to this method

File: mcs/class/Mono.Profiler.Log/Mono.Profiler.Aot/ProfileWriter.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+			foreach (var t in data.Types)
+				AddType (t);
+
+			// just to be complete, do not miss any module too
+			foreach (var module in data.Modules)
+				AddModule (module);
+
+			WriteRecord (RecordType.NONE, 0);
+
+			s.Flush ();
```
Comment (marek-safar) [inline mcs/class/Mono.Profiler.Log/Mono.Profiler.Aot/ProfileWriter.cs]: Why is this needed?

**Comment by radekdoulik** [comment]
Updated

**Comment by radekdoulik** [comment]
@monojenkins build failed

**Comment by radekdoulik** [comment]
@monojenkins build failed
